src/Riok.Mapperly/Riok.Mapperly.Roslyn4.8.props

```diff
@@ -0,0 +1,12 @@
+<Project>
+
+  <PropertyGroup>
+    <DefineConstants>$(DefineConstants);ROSLYN4_4_OR_GREATER</DefineConstants>
+    <DefineConstants>$(DefineConstants);ROSLYN4_8_OR_GREATER</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.8.0-2.final" PrivateAssets="all" />
```

Comment:
If 4.8 is really needed can you also update the default version in `Riok.Mapperly.csproj > ROSLYN_VERSION`.
Can you explain to me why 4.8 is required instead of 4.7?

Reply:
I was literally just about to ask for help with props/preprocessors, before you added your comments. I'm struggling
😆 

> If 4.8 is really needed 

Not 100% sure but here's my thinking: `DescriptorBuilder` should emit a diagnostic for versions of .NET lower than 8, to this I used `#if ROSLYN4_4_OR_GREATER`. This worked until I realised that the integration tests would fail because Roslyn 4.4 applied to older versions of .NET.

I tried to fix this by adding `ROSLYN4_8_OR_GREATER` which requires its corresponding props. As I understand it, Roslyn 4.8 corresponds to .NET 8. I could set the version for `Microsoft.CodeAnalysis.CSharp` to 4.7.0 but I thought it made more sense to use 4.8.0.

I'd have preferred not to make any of these changes, but I can't think of how else to add a new constant. I'd appreciate guidance here, I've never really understood custom constants and props.

Reply:
Since this PR is already huge, just use `4.7` for the condition, this should work for now. We can then add `4.8` in another PR (the steps needed are more or less explained [here](https://mapperly.riok.app/docs/contributing/common-tasks/#add-support-for-a-new-roslyn-version)). Let me know if you need any help on how to do this.